pkg/util/pullsecret/pullsecret.go

```diff
@@ -164,3 +169,41 @@ func Build(oc *api.OpenShiftCluster, ps string) (string, error) {
 
 	return pullSecret, nil
 }
+
+type RegistryClient struct {
+	CheckAuth func(context.Context, *types.SystemContext, string, string, string) error
+}
+
+func NewRegistryClient() RegistryClient {
+	return RegistryClient{
+		CheckAuth: dockerregistry.CheckAuth,
+	}
+}
+
+// ValidatePullSecret validates a passed in pull secret by attempting to log in to the registry
+func (r *RegistryClient) ValidatePullSecret(ctx context.Context, secret *corev1.Secret) error {
+	dockerConfig, err := UnmarshalSecretData(secret)
+	if err != nil {
+		return err
+	}
+	for registry, authBase64 := range dockerConfig {
+		authDecoded, err := base64.StdEncoding.DecodeString(authBase64)
+		if err != nil {
+			return err
+		}
+
+		auth := strings.SplitN(string(authDecoded), ":", 2)
+		if len(auth) != 2 {
+			err = fmt.Errorf("credentials format error: %s", registry)
+			return err
```

Comment:
Nit - might as well consolidate these lines since you did it on line 205:
```suggestion
			return fmt.Errorf("credentials format error: %s", registry)
```

Reply:
+1